torch/nn/_functions/thnn/fold.py

```diff
@@ -0,0 +1,91 @@
+from torch.autograd.function import Function, once_differentiable
+from torch._thnn import type2backend
+
+from . import _all_functions
+
+
+class Col2Im(Function):
+
+    @staticmethod
+    def forward(ctx, input, output_size, kernel_size, dilation, padding, stride):
+
+        ctx.output_size = output_size
+        ctx.kernel_size = kernel_size
+        ctx.dilation = dilation
+        ctx.padding = padding
+        ctx.stride = stride
+
+        ctx.save_for_backward(input)
```

Comment:
Do you have to save input for backward()? I think all you need is input's size.

Reply:
I think you're right - this is unnecessary - I will remove it

Reply:
Done